front-end/src/components/Post/PostReferendumInfo.tsx

```diff
@@ -21,42 +23,77 @@ const PostReferendumInfo = ({ className, onchainLink }: Props) => {
 
 	return (
 		<div className={className}>
-			{proposerAddress &&
-				<>
-					Proposer: {proposerAddress} <br/>
-				</>
-			}
-			{depositAmount && method && preimageArguments && metaDescription &&
-			<>
-				Deposit: {depositAmount} KSM<br/>
-				Description: {metaDescription}<br/>
-				Method: {method}<br/>
-				{preimageArguments && preimageArguments.length
-					? <>Arguments:
-						<div className={'methodArguments'}>
+			<h4>On-Chain Info</h4>
+			{proposerAddress && depositAmount && method && preimageArguments && metaDescription &&
```

Comment:
Same here, we should have either the proposer as the first argument and then a blanket check for all the element. Or the best option, a check for each element before showing it.